pkg/watchdog/pkg_test.go

```diff
@@ -4,8 +4,374 @@
 
 package watchdog
 
-import "testing"
+import (
+	"fmt"
+	"io/ioutil"
+	"testing"
+	"time"
 
-func TestTODO(t *testing.T) {
-	// TODO: Write a unit test.
+	"golang.org/x/sys/unix"
+)
+
+func getTestStruct() (*Watchdog, error) {
+	f, err := ioutil.TempFile("", "testfile-")
+	if err != nil {
+		return nil, err
+	}
+	return &Watchdog{
+		File:               f,
+		unixSyscall:        nil,
+		unixIoctlGetUint32: nil,
+	}, nil
+}
+
+func TestOpen(t *testing.T) {
```

Comment:
I'm a bit unclear on the purpose of this test? is it testing anything that is logically part of the watchdog?

Reply:
I'll remove it